src/search_index.c

```diff
@@ -0,0 +1,24 @@
+/** .
+ * @file search_index.c
+ * for doxy
+ *
+ *
+ */
+#include <stdio.h>
+#include <stdlib.h>
+
+//**input stores array values in *input[0 - x] 
+//i.e. 32 Number Array is accessable through *(input[index] + x) | with index = 1
+//To create a **input which can be used for arr_long/short a generic algorithm like
+//search_index is used to return index without the need to specify long or short index
+int search_index(int num){
+
+    int arr_long[] = {8, 32, 128, 512, 2048, 8192, 32768};
+    int arr_short[] = {8, 16, 64};
```

Comment:
Das ist der search index der "berechnet" in welchem index vom **input sich das jeweilige array befindet. So eine Realisierung ermöglicht eine viel größere Flexibilität und wiederverwendbarkeit von **input -> Speicherleaks werden reduziert.